enhancements/upgrade-readiness.md

```diff
@@ -0,0 +1,536 @@
+- Preserve existing behavior should the operator not communicate its upgradeable status.
+
+### Non-Goals
+
+- Provide operators with the means to cancel an upgrade that is in process.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As an operator author, I want my operator to be able to communicate conditions to OLM that it cannot infer otherwise.
+
+#### Story 2
+
+As an operator author, I want my operator to be able to communicate that OLM should not attempt to upgrade or uninstall my operator until I tell it otherwise.
+
+## Design Details
+
+OLM will introduce the `Probe` CustomResourceDefinition which will communicate an operator's state through the use of [conditions](https://dev.to/maelvls/what-the-heck-are-kubernetes-conditions-for-4je7). For those unfamiliar with conditions, they are used by many Kubernetes Resources to provide users with additional insight into the state of the resource and typically adhere to the following format:
+
+| Field | Description |
+| ----- | ----------- |
+| Type |  A one-word description of the condition. |
+| Status | Status is the status of the condition. Can be True, False, Unknown. |
+| Reason | Unique, one-word, CamelCase reason for the condition's last transition. |
+| Message | Human-readable message indicating details about last transition. Typically used by cluster admins for debugging purposes. |
+| LastTransitionTime | Last time the condition transitioned from one status to another. |
+
+OLM will define a set of "OLM Supported Conditions" which will influence the behavior of OLM when present. The list of "OLM Supported Conditions" can be seen in the table below:
+
+| Condition Type | Effect |
+| -------------- | ------ |
+| Upgradeable | When `false`, OLM will not upgrade the operator. The pod associated with the operator may still be killed by external forces.|
+| Important | OLM will highlight this condition to the Cluster Admin.|
+
+Operators will communicate thier conditions through one of two methods:
+
+1. By setting conditions on CustomResource that the operator owns.
+2. By setting conditions on a `Probe` CustomResource directly.
+
+#### Why allow operators to communicate status through CustomResources
+
+There are many reasons why operators should communicate their conditions through owned CustomResources:
+
+- CustomResources are the main entrypoint to an operator and operators are already expected to update the status of the CustomResource to communicate "Actual State".
+- By writing the condition to the status of an owned CustomResource, it becomes very clear which CR is responsible for changing OLM behavior.
+- Introducing an OLM owned CRD forces operators to integrate with OLM at a much deeper level than writing to a resource that they already manage. Operators would be required to detect if it is being managed by OLM and begin writing to the API. By writing to an owned CR, the Operator does not require OLM specific code and can be run independently of OLM.
+
```

Comment:
OLM will start a SPO, which will manage the resource. The SPO will not need to create any pods. I will update this tonight.